zombienet_tests/functional/0001-parachains-pvf.zndsl

```diff
@@ -33,14 +33,14 @@ one: parachain 2006 is registered within 60 seconds
 two: parachain 2007 is registered within 60 seconds
 
 # Check if network is fully connected.
-alice: reports peers count is at least 15 within 20 seconds
-bob: reports peers count is at least 15 within 20 seconds
-charlie: reports peers count is at least 15 within 20 seconds
-dave: reports peers count is at least 15 within 20 seconds
-ferdie: reports peers count is at least 15 within 20 seconds
-eve: reports peers count is at least 15 within 20 seconds
-one: reports peers count is at least 15 within 20 seconds
-two: reports peers count is at least 15 within 20 seconds
+alice: reports peers count is at least 8 within 20 seconds
```

Comment:
I'm confused, we have here 16 peers and all of them should be connected to each other? Why are we now only testing for `8` which is the number of outgoing connections? Or do we block incoming connections in zombienet?

Reply:
Previously the number of outbound connection was 15 so the test verified that it was able to establish all 15 outbound connections. Now that number is 8 but the assumption it's testing is still the same. Estimating how many inbound connections each node will get is harder. 

Like I mentioned in Matrix, if we want all of the nodes to be connected to each other, we can also change the `--out-peers` parameter in these tests.

Reply:
Zombienet don't block any traffic between nodes in the same `namespace` (each test case).

Reply:
It seems that at least for the paritydb test we need to increase the amount of outbound connections

Reply:
> Previously the number of outbound connection was 15 so the test verified that it was able to establish all 15 outbound connections.

Wait, I would have thought that "peers count" == "inbound + outbound peers from sync"?

Reply:
> > Previously the number of outbound connection was 15 so the test verified that it was able to establish all 15 outbound connections.
> 
> Wait, I would have thought that "peers count" == "inbound + outbound peers from sync"?

If there are fewer outbound slots than number of peers on the network, the number of total peers is non-deterministic because we can't determine a priori which are the connections a certain node decides to make and thus how many actual connections each node has.

But because the number of inbound slots is more than the number of nodes on the network, we know that each node can establish at least `out-peers` many connections so the test is verifying that it was able to establish all the outbound connections it wanted + any incoming connections from other peers.

Reply:
But over time all peers should be connected to each other? We don't keep bi-directional connections. Meaning if A and B connect to each other with outgoing connections, one connection should be closed over time and then one of them should look for another peer. But yeah, we can not and we should not wait for this in this tests. 

I think we should remove all the "reports peers count" lines in all Zombienet tests in Polkadot. None of these tests is testing sync related things where we would need to check for these metrics. All these tests here are checking other conditions to ensure that progress is made. If you wanted to know if a node is following the chain, you need to check that it imports new blocks and to how many nodes they are connected is from zero importance.

Reply:
> But over time all peers should be connected to each other? We don't keep bi-directional connections. Meaning if A and B connect to each other with outgoing connections, one connection should be closed over time and then one of them should look for another peer.

Not necessarily, right? For simplicity's sake, let's assume each node is only able to establish 3 outbound connections and infinite inbound connections. If there are 6 nodes on the network (node1, ..., node6). Node1 establishes outbound connections to node2, node3, and node4 and it has no more outbound slots. Node6 also establishes outbound connections to these nodes (2, 3, and 4). It also has no more outbound connections and hence node1 and node6 are not directly connected but as long as gossiping works, node6 is able to learn of anything node1 might publish because they're connected through nodes 2, 3 and 4.

I agree with removing the peer count checks though.

Reply:
> I agree with removing the peer count checks though.

So can you please remove them in all files? And also revert your `--out-peers` increase?

Reply:
I think the `out-peers` change is required because the paritydb test failed in Authority discovery or something.

Reply:
> I think the `out-peers` change is required because the paritydb test failed in Authority discovery or something.

Sounds like another bug? :P Authority discovery works using DHT/Kademlia and shouldn't be influenced by the number of out peers in sync.

Reply:
You're right. I reverted that change as well, let's see what CI says.

Reply:
Now CI is green, but maybe at some point it will fail again :D But yeah, then let's see later what the problem is.

Reply:
> But over time all peers should be connected to each other? We don't keep bi-directional connections. Meaning if A and B connect to each other with outgoing connections, one connection should be closed over time and then one of them should look for another peer. But yeah, we can not and we should not wait for this in this tests.
> 
> I think we should remove all the "reports peers count" lines in all Zombienet tests in Polkadot. None of these tests is testing sync related things where we would need to check for these metrics. All these tests here are checking other conditions to ensure that progress is made. If you wanted to know if a node is following the chain, you need to check that it imports new blocks and to how many nodes they are connected is from zero importance.

I agree that the test would fail anyway, but failing earlier is better because it is easier to debug. You would immediately know there is a connectivity issue.

Reply:
yeah it could be that I saw something that wasn't an actual error, I'll merge this and wait to see what pops up

Reply:
> I agree that the test would fail anyway, but failing earlier is better because it is easier to debug. You would immediately know there is a connectivity issue.

If there is a connectivity issue, stuff is broken in Substrate already in sync and it should be detected there and not here.